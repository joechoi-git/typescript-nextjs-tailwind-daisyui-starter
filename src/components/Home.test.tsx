import React from "react";
import { render, screen } from "@testing-library/react";
import Home from "./Home";

// Mock the dynamic import for Grid component
jest.mock("next/dynamic", () => {
    return jest.fn().mockImplementation((importCallback, options) => {
        const Component = importCallback();
        Component.displayName = "Grid";
        return function DynamicComponent(props: React.JSX.IntrinsicAttributes) {
            if (options?.loading) {
                return options.loading();
            }
            return <Component {...props} />;
        };
    });
});

// eslint-disable-next-line react/display-name
jest.mock("next/image", () => (props: { src: string; alt: string; [key: string]: any }) => {
    const { src, alt, ...rest } = props;
    // eslint-disable-next-line @next/next/no-img-element
    return <img src={src} alt={alt} {...rest} />;
});

describe("Home", () => {
    it("renders the article content", () => {
        render(<Home />);
        expect(
            screen.getByText("Garlic bread with cheese: What the science tells us")
        ).toBeInTheDocument();
        expect(screen.getByText(/For years parents have espoused/i)).toBeInTheDocument();
        expect(screen.getByText(/But a recent study shows/i)).toBeInTheDocument();
    });

    it("renders the Buttons component", () => {
        render(<Home />);
        expect(screen.getByText("button 1")).toBeInTheDocument();
    });

    it("renders the Grid component", () => {
        render(<Home />);
        expect(screen.getByText("Loading...")).toBeInTheDocument();
    });

    it("renders the Progress components", () => {
        render(<Home />);
        expect(screen.getByText("0%")).toBeInTheDocument();
        expect(screen.getByText("20%")).toBeInTheDocument();
        expect(screen.getByText("40%")).toBeInTheDocument();
        expect(screen.getByText("60%")).toBeInTheDocument();
        expect(screen.getByText("80%")).toBeInTheDocument();
        expect(screen.getByText("100%")).toBeInTheDocument();
    });

    it("renders the carousel images", () => {
        render(<Home />);
        expect(screen.getByAltText("1")).toBeInTheDocument();
        expect(screen.getByAltText("2")).toBeInTheDocument();
        expect(screen.getByAltText("3")).toBeInTheDocument();
        expect(screen.getByAltText("4")).toBeInTheDocument();
    });

    it("renders the carousel navigation buttons", () => {
        render(<Home />);
        expect(screen.getAllByRole("link", { name: "1" })).toHaveLength(1);
        expect(screen.getAllByRole("link", { name: "2" })).toHaveLength(1);
        expect(screen.getAllByRole("link", { name: "3" })).toHaveLength(1);
        expect(screen.getAllByRole("link", { name: "4" })).toHaveLength(1);
    });

    it("renders the table with correct data", () => {
        render(<Home />);
        expect(screen.getByText("Cy Ganderton")).toBeInTheDocument();
        expect(screen.getByText("Hart Hagerty")).toBeInTheDocument();
        expect(screen.getByText("Brice Swyre")).toBeInTheDocument();
    });

    it("matches the snapshot", () => {
        const { asFragment } = render(<Home />);
        expect(asFragment()).toMatchSnapshot();
    });
});
