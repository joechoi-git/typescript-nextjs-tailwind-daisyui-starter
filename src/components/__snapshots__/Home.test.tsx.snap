// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches the snapshot 1`] = `
<DocumentFragment>
  <article
    class="prose !max-w-none"
  >
    <h1>
      Garlic bread with cheese: What the science tells us
    </h1>
    <p>
      For years parents have espoused the health benefits of eating garlic bread with cheese to their children, with the food earning such an iconic status in our culture that kids will often dress up as warm, cheesy loaf for Halloween.
    </p>
    <p>
      But a recent study shows that the celebrated appetizer may be linked to a series of rabies cases springing up around the country.
    </p>
  </article>
  <section
    class="py-4"
  >
    <p>
      Clicked 0 time.
    </p>
    <div
      class="flex w-full py-2 gap-2"
    >
      <button
        class="btn btn-primary "
      >
        button 1
      </button>
      <button
        class="btn btn-primary "
      >
        button 2
      </button>
      <button
        class="btn btn-primary "
      >
        button 3
      </button>
    </div>
  </section>
  <div>
    Loading...
  </div>
  <section
    class="flex p-4"
  >
    <div
      class="radial-progress bg-primary text-primary-content border-4 border-primary"
      role="progressbar"
      style="--value: 0;"
    >
      0%
    </div>
    <div
      class="radial-progress bg-primary text-primary-content border-4 border-primary"
      role="progressbar"
      style="--value: 20;"
    >
      20%
    </div>
    <div
      class="radial-progress bg-primary text-primary-content border-4 border-primary"
      role="progressbar"
      style="--value: 40;"
    >
      40%
    </div>
    <div
      class="radial-progress bg-primary text-primary-content border-4 border-primary"
      role="progressbar"
      style="--value: 60;"
    >
      60%
    </div>
    <div
      class="radial-progress bg-primary text-primary-content border-4 border-primary"
      role="progressbar"
      style="--value: 80;"
    >
      80%
    </div>
    <div
      class="radial-progress bg-primary text-primary-content border-4 border-primary"
      role="progressbar"
      style="--value: 100;"
    >
      100%
    </div>
  </section>
  <div
    class="carousel w-full p-4"
  >
    <div
      class="carousel-item w-full"
      id="item1"
    >
      <img
        alt="1"
        class="w-full"
        height="250"
        src="https://img.daisyui.com/images/stock/photo-1625726411847-8cbb60cc71e6.jpg"
        width="1000"
      />
    </div>
    <div
      class="carousel-item w-full"
      id="item2"
    >
      <img
        alt="2"
        class="w-full"
        height="250"
        src="https://img.daisyui.com/images/stock/photo-1609621838510-5ad474b7d25d.jpg"
        width="1000"
      />
    </div>
    <div
      class="carousel-item w-full"
      id="item3"
    >
      <img
        alt="3"
        class="w-full"
        height="250"
        src="https://img.daisyui.com/images/stock/photo-1414694762283-acccc27bca85.jpg"
        width="1000"
      />
    </div>
    <div
      class="carousel-item w-full"
      id="item4"
    >
      <img
        alt="4"
        class="w-full"
        height="250"
        src="https://img.daisyui.com/images/stock/photo-1665553365602-b2fb8e5d1707.jpg"
        width="1000"
      />
    </div>
  </div>
  <div
    class="flex justify-center w-full py-2 gap-2"
  >
    <a
      class="btn btn-xs"
      href="#item1"
    >
      1
    </a>
    <a
      class="btn btn-xs"
      href="#item2"
    >
      2
    </a>
    <a
      class="btn btn-xs"
      href="#item3"
    >
      3
    </a>
    <a
      class="btn btn-xs"
      href="#item4"
    >
      4
    </a>
  </div>
  <div
    class="overflow-x-auto p-4"
  >
    <table
      class="table"
    >
      <thead
        class="bg-base-200"
      >
        <tr>
          <th />
          <th>
            Name
          </th>
          <th>
            Job
          </th>
          <th>
            Favorite Color
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th>
            1
          </th>
          <td>
            Cy Ganderton
          </td>
          <td>
            Quality Control Specialist
          </td>
          <td>
            Blue
          </td>
        </tr>
        <tr>
          <th>
            2
          </th>
          <td>
            Hart Hagerty
          </td>
          <td>
            Desktop Support Technician
          </td>
          <td>
            Purple
          </td>
        </tr>
        <tr>
          <th>
            3
          </th>
          <td>
            Brice Swyre
          </td>
          <td>
            Tax Accountant
          </td>
          <td>
            Red
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
